### ISSUE TITLE
Question about accessibility options

### ISSUE THREAD

**Issue by liamzee** [libsdl-org/SDL#8210]
Hi, I'm a user of SDL2 via Haskell's monomer GUI library, and I've talked to others about using monomer for projects (monomer is singly-maintained and supported, so I think it'll go out of use eventually).

One major issue of SDL2 vs GTK is that SDL2 doesn't come with accessibility bindings; i.e, text-to-speech and other hook-ins. Is this EXCLUDED from the goals of SDL2; i.e, if I were to PR accessibility bindings into SDL2, would the SDL2 project have issues with such?

**Comment by smcv** [comment]
Is SDL sufficiently high-level for accessibility bindings to be useful? GTK accessibility deals with concepts like "this is a menu", "this is a button", "the text of this widget is `Cancel`" and so on, but at the SDL level, UIs are just an undifferentiated blob of pixels (it doesn't even have text rendering, you need an addon like SDL_ttf for that).

In terms that a GTK user would understand, SDL is closer to the level of [GDK](https://docs.gtk.org/gdk3/) than GTK (not to be confused with [Microsoft's GDK](https://learn.microsoft.com/en-us/gaming/gdk/), which is unrelated).

If monomer builds a UI framework with buttons, menus and so on in terms of SDL, I would have expected that it would be monomer and not SDL that would need to talk to accessibility tools, because SDL doesn't have enough information.

**Comment by icculus** [comment]
@smcv is right, but if someone wants to propose an API that can be implemented without it becoming a massive undertaking better handled at a different layer, I'm willing to listen.

But until then, I'm going to close this under the assumption it isn't going to happen. We can easily reopen if someone wants to send a PR or even just pitch an idea in a comment below.
